File: backend/src/main/java/com/vinicius/membermanagement/entity/Member.java
package com.vinicius.membermanagement.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "members")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String cpf;

    @Column(nullable = false)
    private LocalDate birthDate;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;
}

File: frontend/src/types/member.ts
export interface Member {
  id: number;
  name: string;
  cpf: string;
  birthDate: string;
  active: boolean;
}

export interface MemberRequest {
  name: string;
  cpf: string;
  birthDate: string;
  active: boolean;
}

export interface ApiError {
  field?: string;
  message: string;
}
